feat: change user property name to createdBy

vietnguyen opened the pull request:
- Removed the HibernateListener solution as it may cause performance issue.
- Added code to ignore `createdBy` field in `IdentifiableObjectBundleHook.preUpdate`
- Test covered by `MetadataImportServiceTest#testUpdateImmutableCreatedByField`

<img width="390" alt="Screen Shot 2021-01-13 at 7 23 50 PM" src="https://user-images.githubusercontent.com/766102/104452000-ef3b4a00-55d4-11eb-812b-b62a93ed28b3.png">

stian-sandvold commented:
Good work @vietnguyen . I see a few automatic comments from github in the files, did you have a look at them? Some seem sensible to check out.

There is also 1 "critical" issue from SonarCloud check, related to using a constant for "createdBy" in one of the files, would be great to add that to please sonarCloud :)

larshelge commented:
Let's make the `createdBy` property unmodifiable / immutable.

larshelge reviewed (changes requested):
Let's make the `createdBy` field immutable / not modifiable.

larshelge commented:
Remember to upgrade the change note and share with front-end team.

larshelge commented on dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/adapter/BaseIdentifiableObject_.java:
```diff
@@ -0,0 +1,65 @@
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.common.adapter;/*
+                                     * Copyright (c) 2004-2021, University of Oslo
```
Remove license.

vietnguyen replied:
fixed

vietnguyen commented:
> Let's make the `createdBy` field immutable / not modifiable.
@larshelge 
- Removed the HibernateListener solution as it may cause performance issue.
- Added code to ignore `createdBy` field in `IdentifiableObjectBundleHook.preUpdate`
- Test covered by `MetadataImportServiceTest#testUpdateImmutableCreatedByField`
